components/ArtPieceDetails.js

```diff
@@ -0,0 +1,35 @@
+import Image from "next/image";
+import styled from "styled-components";
+
+const ImageWrapper = styled.div`
+  display: flex;
+  position: relative;
+  width: 300px;
+  height: 300px;
+`;
+export default function ArtPieceDetails({
+  image,
+  title,
+  artist,
+  year,
+  genre,
+  onClickBack,
+}) {
+  return (
+    <>
+      <ImageWrapper>
+        <Image
+          src={image}
+          alt={title}
+          fill
+          sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 200px"
+        />
+      </ImageWrapper>
+      <h3>{artist}</h3>
+      <p>{year}</p>
+      <p>{genre}</p>
+
+      <button onClick={onClickBack}>Go back</button>
```

Comment:
Instead of using an imperative `router.push`, you could use a `<Link>` for navigation.

This approach is better because it provides prefetching, improves performance and adheres to declarative programming principles.

Example:
```js
<Link href="/art-pieces"> Back to list </Link>
```

Reply:
Thank you , In the beginning we did that but depending on what is in the tasks we should use a back button component.. :)